### PR TITLE
test(osm-bootstrap): add unit tests for mesh config

### PR DESCRIPTION
**Pull request by michelleN**
+ adds unit tests for mesh config logic

Signed-off-by: Michelle Noorali <minooral@microsoft.com>

**Description**:

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

### CODE REVIEW

File: cmd/osm-bootstrap/osm-bootstrap_test.go [hunk 1]
```diff
@@ -1,67 +1,86 @@
...
+	Data: map[string]string{
+		presetMeshConfigJSONKey: `{
 "sidecar": {
-  "enablePrivilegedInitContainer": false,
-  "logLevel": "error",
-  "maxDataPlaneConnections": 0,
-  "envoyImage": "envoyproxy/envoy-alpine@sha256:6502a637c6c5fba4d03d0672d878d12da4bcc7a0d0fb3f1d506982dde0039abd",
-  "initContainerImage": "openservicemesh/init:latest-main",
-  "configResyncInterval": "2s"
+"enablePrivilegedInitContainer": false,
```
Comment (allenlsy) [inline cmd/osm-bootstrap/osm-bootstrap_test.go:41]: Indentation was removed?

**Review by allenlsy — APPROVED** [review]

File: cmd/osm-bootstrap/osm-bootstrap_test.go (hunk 1 above)
Comment (michelleN) [reply cmd/osm-bootstrap/osm-bootstrap_test.go]: didn't impact tests but I added back the indentation for readability. thanks for noticing.

File: cmd/osm-bootstrap/osm-bootstrap.go [hunk 2]
```diff
@@ -211,6 +211,45 @@ func main() {
...
+
+	// Create a default meshConfig
+	defaultMeshConfig := buildDefaultMeshConfig(presetsConfigMap)
+	if _, err := b.meshConfigClient.ConfigV1alpha1().MeshConfigs(b.namespace).Create(context.TODO(), defaultMeshConfig, metav1.CreateOptions{}); err == nil {
+		log.Info().Msgf("MeshConfig (%s) created in namespace %s", meshConfigName, b.namespace)
+		return nil
+	}
+
+	if apierrors.IsAlreadyExists(err) {
+		return nil
```
Comment (snehachhabria) [inline cmd/osm-bootstrap/osm-bootstrap.go:231]: Could we add the logs that were present previously

**Review by snehachhabria — APPROVED** [review]

File: cmd/osm-bootstrap/osm-bootstrap.go (hunk 2 above)
Comment (michelleN) [reply cmd/osm-bootstrap/osm-bootstrap.go:231]: I just bubbled up the error to main() where there is one log line which wraps the error https://github.com/openservicemesh/osm/pull/4181/files#diff-2aa329ee12aeb0b3d3a60599559a0bbe07422779d9f730f4be4e668b275e06acR146

I thought a log line here might be confusing but I can add it in.

Comment (michelleN) [reply cmd/osm-bootstrap/osm-bootstrap.go:232]: added.

**Pull request merged**